src/Mimi.jl

```diff
@@ -216,14 +216,15 @@ function setparameter(m::Model, component::Symbol, name::Symbol, value)
 end
 
 function checklabels(m::Model, component::Symbol, name::Symbol, parametername::Symbol, p::ArrayModelParameter)
-    if !(eltype(p.values) <: getmetainfo(m, component).parameters[parametername].datatype)
+    meta_comp_name = Symbol(string(m.components2[component].component_type))
+    if !(eltype(p.values) <: getmetainfo(m, meta_comp_name).parameters[parametername].datatype)
```

Comment:
I think the error is here. I think you need to call ``getmetainfo`` with the name of the component in the model, not the type of the component. But the way you construct ``meta_comp_name``, it will be the name of the type. I think you just need to change this to ``getmetainfo(m, component)``, and it should work?